toolchain/check/testdata/interop/cpp/macros.carbon

```diff
@@ -139,6 +139,36 @@ fn F() {
   let b: i32 = Cpp.N.CONFIG_VALUE;
 }
 
+// --- macro_and_non_macro_same_name.h
+
+namespace X {
+  float n;
+}
+float n;
+#define n 1
+
+// --- fail_import_macro_and_non_macro_same_name.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "macro_and_non_macro_same_name.h";
+
+fn F() {
+  // When the same macro name and non-macro name exist in the global C++ scope, the macro name is selected.
+  let a: i32 = Cpp.n;
+
+  // CHECK:STDERR: fail_import_macro_and_non_macro_same_name.carbon:[[@LINE+7]]:16: error: cannot implicitly convert expression of type `i32` to `f32` [ConversionFailure]
+  // CHECK:STDERR:   let b: f32 = Cpp.n;
+  // CHECK:STDERR:                ^~~~~
+  // CHECK:STDERR: fail_import_macro_and_non_macro_same_name.carbon:[[@LINE+4]]:16: note: type `i32` does not implement interface `Core.ImplicitAs(f32)` [MissingImplInMemberAccessNote]
+  // CHECK:STDERR:   let b: f32 = Cpp.n;
+  // CHECK:STDERR:                ^~~~~
+  // CHECK:STDERR:
+  let b: f32 = Cpp.n;
```

Comment:
We should eventually allow an implicit conversion from `i32` to `f32`, at which point I think this test won't test that the macro is being selected. How about something like:

```suggestion
  var b: array(f32, Cpp.n) = (1.0,);
```

... to test that it's being treated as an integer constant with value 1.